src/main/src/main/java/org/geoserver/config/impl/GeoServerInfoImpl.java

```diff
@@ -40,6 +40,8 @@
 
     protected Boolean globalServices = true;
 
+    protected Boolean useHeadersProxyURL = false;
```

Comment:
This is guaranteed to be null when XStream deserializes from XML, if the element is missing, as it won't go through normal class construction (so forget about constructors and field initializers, that are also invoked as part of normal constructor invocation).

Reply:
OK, I will solve this tomorrow.

Reply:
Hopefully this is solved: please, refer to this other PR: #2968